### PR TITLE
The users app refactored. JWT utilities implemented under common package

### PR DESCRIPTION
**Pull request by ankushchavan1**
1. JWT utilities module implemented under common package.
2. JWT token generation and token verifiication tasks separated from apps.
3. The users app refactored.

### CODE REVIEW

File: server/apps/todo/models.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+from django.db import models
+from django.utils.timezone import now
+from model_utils import Choices
+from ..common.model_utils import TimestampedModel, CreatorInfoModel
+
+from ..users.models import User
```
Comment (ronakmutha) [inline server/apps/todo/models.py]: Use relative imports here

**Review by ronakmutha — CHANGES_REQUESTED** [review]

File: server/apps/common/model_utils.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+class TimestampedModel(models.Model):
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    class Meta:
+        abstract = True
+
+
+class CreatorInfoModel(models.Model):
```
Comment (ronakmutha) [inline server/apps/common/model_utils.py]: You can all this Audit Model

File: server/apps/common/jwt_utils.py [hunk 3]
```diff
@@ -30,3 +31,25 @@ def generate_jwt_refresh_token(user):
...
     return refresh_token
+
+
+def verify_jwt_token(token):
+    if not token:
+        raise NotAuthenticated('Unauthenticated!')
+    try:
+        payload = jwt.decode(token, settings.SECRET_KEY, algorithms=['HS256'])
+    except jwt.ExpiredSignatureError:
+        raise NotAuthenticated('Authentication failed! Signature Expired!')
```
Comment (ronakmutha) [inline server/apps/common/jwt_utils.py]: All these strings should be from a constants file.
You can create a python file `messages.py`. And add in the strings there

File: server/apps/todo/admin.py [hunk 4]
```diff
@@ -0,0 +1,4 @@
+from django.contrib import admin
+from . import models
+
+admin.site.register(models.Todo)
```
Comment (ronakmutha) [inline server/apps/todo/admin.py]: You don't require admin portal

File: server/apps/todo/models.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+
+class Todo(TimestampedModel, CreatorInfoModel):
+    COMPLETED = "completed"
+    INPROGRESS = "inprogress"
+    STATUS = Choices((COMPLETED, COMPLETED), (INPROGRESS, INPROGRESS))
+    status = models.CharField(choices=STATUS, default=INPROGRESS, max_length=20)
+    owner = models.CharField(max_length=100)
```
Comment (ronakmutha) [inline server/apps/todo/models.py]: Owner would be a User

File: server/apps/todo/views.py [hunk 6]
```diff
@@ -0,0 +1,28 @@
+from django.db.models import Q
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from .models import Todo
+from .serializer import TodoSerializer
+from ..common.jwt_utils import verify_jwt_token
+from ..users.models import User
```
Comment (ronakmutha) [inline server/apps/todo/views.py]: Incorrect way to add imports

File: server/apps/users/views.py [hunk 7]
```diff
@@ -1,15 +1,13 @@
...
-from rest_framework.exceptions import AuthenticationFailed, NotAuthenticated
+from rest_framework.exceptions import AuthenticationFailed
 from rest_framework.response import Response
 from rest_framework.views import APIView
 
-from .generate_jwt_tokens import generate_jwt_access_token, generate_jwt_refresh_token
+from ..common.jwt_utils import generate_jwt_access_token, generate_jwt_refresh_token
 from .models import User
 from .serializers import UserSerializer
+from ..common.jwt_utils import verify_jwt_token, verify_jwt_refresh_token
```
Comment (ronakmutha) [inline server/apps/users/views.py]: Incorrect imports

**Review by ronakmutha — CHANGES_REQUESTED** [review]

File: server/apps/todo/models.py (hunk 1 above)
Comment (ankushchavan1) [reply server/apps/todo/models.py]: Updated

File: server/apps/todo/admin.py (hunk 4 above)
Comment (ankushchavan1) [reply server/apps/todo/admin.py]: Updated

File: server/apps/todo/views.py (hunk 6 above)
Comment (ankushchavan1) [reply server/apps/todo/views.py]: Updated

File: server/apps/users/views.py (hunk 7 above)
Comment (ankushchavan1) [reply server/apps/users/views.py]: Updated

File: server/apps/common/jwt_utils.py (hunk 3 above)
Comment (ankushchavan1) [reply server/apps/common/jwt_utils.py]: Updated

File: server/apps/common/model_utils.py (hunk 2 above)
Comment (ankushchavan1) [reply server/apps/common/model_utils.py]: Updated

File: server/apps/todo/models.py (hunk 5 above)
Comment (ankushchavan1) [reply server/apps/todo/models.py]: Updated. Changed owner as a ForeignKey to the User model

**Pull request merged**